src/core/mac/sub_mac.cpp

```diff
@@ -934,6 +947,8 @@ void SubMac::SetCslPeriod(uint16_t aPeriod)
         mCslSampleTime = TimerMicro::GetNow();
         Get<Radio>().UpdateCslSampleTime(mCslSampleTime.GetValue());
 
+        mCslLastSync = mCslSampleTime;
```

Comment:
`mCslLastSync` should be set when the first packet with CSL IE include is sent(in `SubMac::HandleTransmitDone`). In this case, a child update request.

Ref: in `Mac::SetCslPeriod` --> `Get<Mle::Mle>().ScheduleChildUpdateRequest()`

Reply:
Now that I capped the maximum duration removing this initialization should be okay.